### PR TITLE
Commando-spark plugin

### PR DESCRIPTION
**Pull request by adi2011**
This will help spark wallet to fetch custom commands remotely. Inspired by @rustyrussell 's commando plugin.

### CODE REVIEW

File: spark-commando/spark-commando.py [hunk 1]
```diff
@@ -0,0 +1,411 @@
...
+        this_rune = plugin.masterrune.copy()
+        this_rune.add_restriction(runes.Restriction.unique_id(plugin.rune_counter))
+    else:
+        this_rune, whynot = is_rune_valid(plugin, rune)
+        if this_rune is None:
+            raise RpcError('commando-rune', {'rune': rune}, whynot)
+
+    read_only_rune = add_reader_restrictions(this_rune)
+
+    restrictions.append("method^spark")
```
Comment (adi2011) [inline spark-commando/spark-commando.py:230]: To restrict user to spark's custom commands only.

File: spark-commando/spark-commando.py [hunk 2]
```diff
@@ -0,0 +1,411 @@
...
+        plugin.rune_counter += 1                
+        plugin.rune_counter_generation += 1
+    
+    #Should replace +=1 to +=2?
+
+    return {'rune': this_rune.to_base64(), 
+            'read-only rune':read_only_rune
+            }
+
+
```
Comment (adi2011) [inline spark-commando/spark-commando.py:253]: Following are the custom commands, will add DOCSTRING after review.

File: spark-commando/spark-commando.py [hunk 3]
```diff
@@ -0,0 +1,411 @@
...
+            try:
+                res = {'result': plugin.rpc.call(method, params)}
+            except RpcError as e:
+                res = {'error': e.error}
+
+    send_result(plugin, peer_id, idnum, res)
+
+
+
+@plugin.hook('custommsg')
```
Comment (adi2011) [inline spark-commando/spark-commando.py:167]: For implementing incoming request? Took directly from commando..

**Comment by adi2011** [comment]
@PestToast  thank you for trying it out, the intended behaviour is that you scan it in the spark wallet (not yet added) and then the spark wallet will be able to control the node(through read-only or fully). I'll definitely print the path string to the rune on the cli thank you for the suggestion.

**Comment by PestToast** [comment]
@adi2011 Thank you for working on this! I hope to get it running soon. Is it working currently on mainnet? My concern is that if I'm running this on a remote server that I ssh into, trying to open it in a web browser will do no good, but if I can pipe the image output somewhere, then I could do whatever I want with it (including opening it in a web browser). Maybe I can find some free time this weekend to look at it closer. I'm not really a developer, but hopefully I'll be able to navigate it enough to figure it out.

**Comment by chrisguida** [comment]
@adi2011 Are you still working on this? I don't think Spark is maintained anymore :/

**Comment by chrisguida** [comment]
Closing as @adi2011 says he is no longer planning to finish this.

**Pull request closed without merging**